priam/src/main/java/com/netflix/priam/defaultimpl/PriamConfiguration.java

```diff
@@ -683,7 +673,7 @@ public boolean isIncrBackup() {
     @Override
     public String getHostIP() {
         if (this.isVpcRing()) return LOCAL_IP;
-        else return PUBLIC_IP;
+        else return getInstanceDataRetriever().getPublicHostname();
```

Comment:
Why not getInstanceDataRetriever().getPublicIP();

Reply:
Good catch, I messed up here.

My primary purpose of doing this is just to defer the call to getPublicHostname/getPublicIP to the time it is actually used. My instances in a private VPC don't have these, so I didn't personally run into an issue.

Would you like this upstream if done correctly? I'd like to search for other usages of instance metadata and isolate into the instanceDataRetriever also to make this easier to run locally with just a different implementation.

Reply:
Good catch, I messed up here.

My primary purpose of doing this is just to defer the call to getPublicHostname/getPublicIP to the time it is actually used. My instances in a private VPC don't have these, so I didn't personally run into an issue.

Would you like this upstream if done correctly? I'd like to search for other usages of instance metadata and isolate into the instanceDataRetriever also to make this easier to run locally with just a different implementation.

Reply:
Yes. I would like this in upstream to make code cleaner and later as you said to run locally. One of the goal for 2018 is to ensure we can run Priam locally for testing.